source/puppet/4.8/env_isolation.md

```diff
@@ -8,25 +8,25 @@ description: "Generating metadata to isolate resources in environments in Puppet
 [code_mgr_env]: {{pe}}/code_mgr.html#environment-isolation-metadata
 
 
-If you have multiple environments, environment isolation prevents resource types from leaking between your various environments. To use environment isolation, you'll generate metadata files that Puppet can use in place of the normal Ruby resource type implementations.
+If you have multiple environments, environment isolation prevents module resource types from leaking between your various environments. To use environment isolation, you'll generate metadata files that Puppet can use in place of the normal Ruby resource type implementations.
```

Comment:
Yeah, I can't say I love it myself. In this paragraph, I need something brief that gives the user an idea of whether this problem is of interest to them. Another option is "custom," but I think that's not great for exactly the reasons you stated.

My inclination at this point is to just leave it as "resource type," and let the more detailed conceptual section expand on that below (not Puppet built-in types, but anything else).